src/OrchardCore/OrchardCore.ContentManagement.Abstractions/Handlers/IContentHandler.cs

```diff
@@ -20,6 +20,8 @@ public interface IContentHandler
         Task ValidatedAsync(ValidateContentContext context);
         Task VersioningAsync(VersionContentContext context);
         Task VersionedAsync(VersionContentContext context);
+        Task DraftSavingAsync(SaveDraftContentContext context);
```

Comment:
Hi @jtkech,

Should these 2 be added to `IContentPartHandler` too?

Btw, we've been using this branch for a while now and we've very happy with it and had no issues.

Reply:
@lazcool Good catch, forgot it ;)

Reply:
Done ;)

> Btw, we've been using this branch for a while now and we're very happy with it and had no issues.

Cool thanks. If you have time can you add some comments on your own usage of this branch in the main thread? It may help us to decide if it is worth to review this PR again before merging, or just close it.

Reply:
Sorry, it's took a week, work's been manic 😅.

We use the new draft saved event, to sync the content item to a preview graph database (we basically replicate content items as nodes and relationships to published and preview graphs).

We'll also be using the new draft saved event to publish external events to Azure Event Grid when a new draft version is saved. (We have micro-service UI apps that listen for the events and then update their local stores with the content from the graphs.)

We currently publish draft events using the existing Orchard Core events, such as updated, but without this PR, we've found it impossible not to publish false-positive events, because the context just isn't there (and we have to do delayed processing of the events, which is complicated, messy and racey).

I have another question for you:
Would it make sense to fire the new draft save events when the user adds or deletes a taxonomy term?

At the moment, as soon as they, for example, add a new term, a new draft version of the taxonomy is saved, but no draft saved event gets fired. That means we don't get an opportunity to sync the new item to our graph or publish an event off the back of it.